src/plugins/Sim.js

```diff
@@ -0,0 +1,303 @@
+'use strict';
+
+const OverlayPlugin = require('./util/OverlayPlugin');
+
+/**
+ * Instance sim plugin.
+ * Provides Sim and operator informations control.
+ */
+module.exports = class Sim extends OverlayPlugin {
+    /**
+     * Plugin initialization.
+     *
+     * @param {Object}  instance        Associated instance.
+     * @param {Object}  i18n            Translations keys for the UI.
+     * @param {boolean} basebandEnabled Whether or not baseband control is enabled.
+     */
+    constructor(instance, i18n, basebandEnabled) {
+        super(instance);
+
+        // Reference instance
+        this.instance = instance;
+
+        // Register plugin
+        this.instance.sim = this;
+        this.i18n = i18n || {};
+
+        this.basebandEnabled = basebandEnabled;
+
+        // Render components
+        this.renderToolbarButton();
+        this.renderWidget();
+
+        // Listen for baseband messages: "<sim/network> <operator/operator_name/imsi_id/phone_number> <value>"
+        this.instance.registerEventCallback('baseband', (message) => {
+            const values = message.split(' ');
```

Comment:
could you create a dedicated callback function instead of defining it here?

Reply:
it's old code moved but you're right it's an opportunity to do cleaner